src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -3185,6 +3192,37 @@ void *SWITCH_THREAD_FUNC sofia_profile_thread_run(switch_thread_t *thread, void
 		profile->tls_verify_in_subjects = su_strlst_dup_split((su_home_t *)profile->nua, profile->tls_verify_in_subjects_str, "|");
 	}
 
+	ssl_ctx = SSL_CTX_new((SSL_METHOD *)ssl_method);
```

Comment:
This must be destroyed when finished.  Any use of SSL must check if `HAVE_OPENSSL` is defined and the correct version of openssl is available.

Reply:
also can we just make this a core helper function that does all what chris said properly and returns a result and use that function here